cardinal/ecs/query_test.go

```diff
@@ -11,41 +11,14 @@ import (
 	"gotest.tools/v3/assert"
 
 	routerv1 "pkg.world.dev/world-engine/rift/router/v1"
-
-	"pkg.world.dev/world-engine/cardinal/ecs"
 )
 
 func TestQueryTypeNotStructs(t *testing.T) {
-	type FooRequest struct {
-		ID string
-	}
-	type FooReply struct {
-		Name string
-		Age  uint64
-	}
-
-	expectedReply := FooReply{
-		Name: "Chad",
-		Age:  22,
-	}
-
-	defer func() {
-		// test should trigger a panic.
-		panicValue := recover()
-		assert.Assert(t, panicValue != nil)
-		ecs.NewQueryType[FooRequest, FooReply]("foo", func(wCtx ecs.WorldContext, req FooRequest) (FooReply, error) {
-			return expectedReply, nil
-		})
-		defer func() {
-			// deferred function should not fail
-			panicValue = recover()
-			assert.Assert(t, panicValue == nil)
-		}()
-	}()
-
-	ecs.NewQueryType[string, string]("foo", func(wCtx ecs.WorldContext, req string) (string, error) {
-		return "blah", nil
+	str := "blah"
+	err := ecs.RegisterQuery[string, string](testutils.NewTestWorld(t).Instance(), "foo", func(wCtx ecs.WorldContext, req *string) (*string, error) {
+		return &str, nil
 	})
+	assert.Assert(t, err != nil)
```

Comment:
we should ensure this error is hitting the right case.

i'd suggest using `assert.ErrorContains` here and make sure it has the `The Request and Reply must be both structs` string.